src/types/iterator.rs

```diff
@@ -97,22 +101,6 @@ impl<'v> PyTryFrom<'v> for PyIterator {
     }
 }
 
-impl Py<PyIterator> {
```

Comment:
These special cases for `PyIterator` etc are no longer needed, because `as_ref()` and `into_ref()` are made available by the `HasPyGilRef` implementation for all `T: HasPyGilRef`.